### ISSUE TITLE
Docker Desktop GUI does not refresh the view

### ISSUE THREAD

**Issue by MBAatEJOT** [docker/for-win#12679]
- [X] I have tried with the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Actual behavior
When deleting a container the list of existings `Containers / App` on the correspondent tab is not refreshing. It's a cointainer created by a `docker-compose` file. So to delete the container itself I have to unfold the `_devcontainer` entry and click on the `delete` button of the actual container. But there is no response that the container is really deleted. No message, no removing the container from the list. It only refreshes the list when I re-open the gui.

### Expected behavior
The removed container should automatically removed by the GUI.

### Information
  - Windows Version: 10 20H2
  - Docker Desktop Version: 4.6.1 (76265)
  - WSL2 or Hyper-V backend?  WSL2
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`

```
[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0031: does the Docker API work?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0001: is the application running?
[SKIP] DD0018: does the host support virtualization?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0017: can a VM be started?
[PASS] DD0024: is WSL installed?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0005: is the user in the docker-users group?
[PASS] DD0007: is the backend responding?
[PASS] DD0014: are the backend processes running?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0006: is the Docker Desktop Service responding?
[PASS] DD0012: is the VM networking working?
[PASS] DD0032: do Docker networks overlap with host IPs?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
No fatal errors detected.
```

### Steps to reproduce the behavior

  1. Create a new devcontainer with vscode and docker-compose.yml
  2. Close VSCode and let the container shut down
  3. Open Docker GUI
  4. unfold the devcontainer entry and delete the actual container
 
docker-compose.yaml example:
```yaml
version: '3'
services:
  app:
    container_name: SimpleTest
    build:
      context: .
      dockerfile: Dockerfile
      args:
        USER_ID: 1000
        USER_GID: 1000
    volumes:
    - /var/run/docker.sock:/var/run/docker-host.sock
    - ..:/workspaces:cached
    # - shareddata:/tmp/host
    network_mode: host
    command: sleep infinity
# volumes:
#   shareddata:
#     external: true
```

Dockerfile Example

```dockerfile
FROM mcr.microsoft.com/vscode/devcontainers/base:0-focal

# Enable immediately list of possible values if terminal input is ambiguous
RUN echo 'set show-all-if-ambiguous on' >> /etc/inputrc
# Run through all possible values by using Shift+Tab
RUN echo '"\\e[Z": menu-complete' >> /etc/inputrc

# Install something
RUN apt-get update \
	&& apt-get install -y wget apt-transport-https curl
```

**Comment by x0rsw1tch** [comment]
Having the same issue. This is occurring on the current version (4.6.1) and the previous version. 4.5.2 didn't have this problem.

I'm on Win10 pro 21H2, with Hyper-V back-end.

**Comment by x0rsw1tch** [comment]
I take that back, it seems to be inconsistent. Just a moment ago, I started up some containers via docker desktop, and the view didn't update. After starting all the containers as "exited" still, even though they are running and functional. After closing and re-opening the window, the view updated.

I stopped the containers and the view updated automatically. After the containers finished shutting down, I started them back up again, and the view self-updated that time, but not the first time. Inconsistent behavior.
